internal/fs/fs.go

```diff
@@ -2637,6 +2638,16 @@ func (fs *fileSystem) WriteFile(
 		ctx, cancel = util.IsolateContextFromParentContext(ctx)
 		defer cancel()
 	}
+
+	if fs.newConfig.Write.ExperimentalEnableRapidAppends {
+		fs.mu.Lock()
+		fh := fs.handles[op.Handle].(*handle.FileHandle)
+		fs.mu.Unlock()
+
+		if err := fh.Write(ctx, op.Data, op.Offset); err != nil {
+			return err
+		}
+	}
```

Comment:
either you need to put the logic from line no: 2651 in else block or add a return statement in the if block